flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/functions/sql/SqlJsonObjectFunction.java

```diff
@@ -42,19 +40,21 @@
 import java.util.Locale;
 
 import static org.apache.calcite.util.Static.RESOURCE;
+import static org.apache.flink.table.planner.plan.type.FlinkReturnTypes.VARCHAR_NOT_NULL;
 
 /**
  * This class has been copied from Calcite to backport the fix made during CALCITE-4394.
  *
- * <p>TODO Remove this class with Calcite 1.27 and replace it with {@link
- * SqlStdOperatorTable#JSON_OBJECT}.
+ * <p>TODO Remove this class with Calcite 1.27 and create a wrapper class for the {@link
+ * SqlJsonObjectFunction} but using the {@code VARCHAR_NOT_NULL} return type inference like {@link
+ * SqlJsonArrayFunctionWrapper}.
  */
 public class SqlJsonObjectFunction extends SqlFunction {
     public SqlJsonObjectFunction() {
         super(
                 "JSON_OBJECT",
                 SqlKind.OTHER_FUNCTION,
-                ReturnTypes.VARCHAR_2000,
+                VARCHAR_NOT_NULL,
```

Comment:
introduce a wrapper for this class already. we should not start patching classes copied from Calcite.